client/src/main/java/io/split/client/impressions/strategy/ProcessImpressionOptimized.java

```diff
@@ -0,0 +1,38 @@
+package io.split.client.impressions.strategy;
+
+import io.split.client.impressions.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class ProcessImpressionOptimized implements ProcessImpressionStrategy{
+
+    private final ImpressionObserver _impressionObserver;
+    private final ImpressionCounter _impressionCounter;
+
+
+    public ProcessImpressionOptimized(ImpressionObserver impressionObserver, ImpressionCounter impressionCounter) {
+        _impressionObserver = impressionObserver;
+        _impressionCounter = impressionCounter;
+    }
+
+    @Override
+    public ImpressionsResult processImpressions(List<Impression> impressions) {
+        List<Impression> impressionsToQueue = new ArrayList<>();
+        for(Impression impression : impressions) {
+            impression = impression.withPreviousTime(_impressionObserver.testAndSet(impression));
+            _impressionCounter.inc(impression.split(), impression.time(), 1);
+            if(!shouldQueueImpression(impression)) {
```

Comment:
Why you are using a negative if?
Personally in this case I prefer to use a positive if and avoid the `continue` statement.

Reply:
I removed the not and changed the logic in the method.